pkg/api/controllers/fileshare.go

```diff
@@ -382,13 +383,22 @@ func (f *FileSharePortal) DeleteFileShareAcl() {
 		return
 	}
 	defer f.CtrClient.Close()
+	metadata := make(map[string]string)
+	for k, v := range fileshare.Metadata {
+		metadata[k] = v
+	}
+
+	for k, v := range acl.Metadata {
+		metadata[k] = v
+	}
+
 	opt := &pb.DeleteFileShareAclOpts{
 		FileshareId:      acl.FileShareId,
 		Description:      acl.Description,
 		Type:             acl.Type,
 		AccessCapability: acl.AccessCapability,
 		AccessTo:         acl.AccessTo,
-		Metadata:         fileshare.Metadata,
+		Metadata:         metadata,
```

Comment:
Please use `utils.MergeStringMaps(fileshare.Metadata, acl.Metadata)` to merge the metadata.